api/types/server.go

```diff
@@ -397,6 +419,64 @@ func (s *ServerV2) setStaticFields() {
 	s.Version = V2
 }
 
+// IsOpenSSHNode returns whether the connection to this Server must use OpenSSH.
+// This returns true for SubKindOpenSSHNode and SubKindOpenSSHEphemeralKeyNode.
+func (s *ServerV2) IsOpenSSHNode() bool {
+	return s.SubKind == SubKindOpenSSHNode || s.SubKind == SubKindOpenSSHEphemeralKeyNode
+}
+
+// openSSHNodeCheckAndSetDefaults are common validations for OpenSSH nodes.
+// They include SubKindOpenSSHNode and SubKindOpenSSHEphemeralKeyNode.
+func (s *ServerV2) openSSHNodeCheckAndSetDefaults() error {
+	if s.Spec.Addr == "" {
+		return trace.BadParameter(`Addr must be set when server SubKind is "openssh"`)
```

Comment:
nit: we usually avoid starting Go errors with uppercase letters.

Same for others.

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Yeah, I saw that as well
This code already existed but I'm going to downcase the starting word :+1:

Reply:
If you could rephrase it so "AWS" is not the first word it would be better.